lib/reporter/JUnitXmlReporter.js

```diff
@@ -0,0 +1,126 @@
+    const nameParts = []
+    nameParts.push(pageResult.pageFilePath)
+    if (pageResult.pageTitle) {
+      nameParts.push(pageResult.pageTitle)
+    } else {
+      nameParts.push(uuidv4())
+    }
+    nameParts.filter(part => part)
+    const name = nameParts.join('_')
+    const sanitizedPath = name.replace(/[^A-Za-z0-9_-]+/g, '_')
+    const testOutputFileName = `TEST-unit-html-${sanitizedPath}.${FILE_EXTENSION}`
+    const testOutputFilePath = path.join(this.outputPath, testOutputFileName)
+
+    // Initialize the XML document
+    const doc = create({ version: '1.0', encoding: 'UTF-8' })
+      .ele('testsuite', {
+        tests: pageResult.nrOfItemsCheckedOnPage().toString(),
+        failures: pageResult.nrOfFindingsOnPage().toString(),
+        errors: '0',
+        time: '0',
+        name: `${pageResult.pageFileName}: ${pageResult.pageTitle}`
+      })
+
+    // Add test cases
+    pageResult.allCheckerResults?.forEach(singleCheckResult => {
+      const testcase = doc.ele('testcase', {
+        assertions: singleCheckResult.nrOfItemsChecked.toString(),
+        time: '0',
+        name: singleCheckResult.whatIsChecked || ''
+      })
+
+      singleCheckResult.findings.forEach(finding => {
+        testcase.ele('failure', {
+          type: singleCheckResult.sourceItemName + ' - ' + singleCheckResult.targetItemName,
+          message: finding.whatIsTheProblem
+        }).txt(finding.suggestions?.join(', ') || '')
+      })
+    })
+
+    // Convert the document to string
+    const xmlString = doc.end({ prettyPrint: true })
+
+    // Write the XML string to a file
+    this.log.info(`Writing JUnit XML report to ${testOutputFilePath}`)
+    fs.writeFileSync(testOutputFilePath, xmlString)
+  }
+
+  reportPageDetails (/* SinglePageResult */ pageResult) {
+    pageResult.allCheckerResults.forEach(resultForOneCheck => {
+      this.reportSingleCheckSummary(resultForOneCheck)
+      this.reportSingleCheckDetails(resultForOneCheck)
+    })
+  }
+
+  reportPageFooter () {
+    // empty in original impl: https://github.com/aim42/htmlSanityCheck/blob/main/src/main/groovy/org/aim42/htmlsanitycheck/report/JUnitXmlReporter.groovy#L87
+  }
+
+  reportSingleCheckSummary (/* SingleCheckResult */ resultForOneCheck) {
+    // empty in original impl: https://github.com/aim42/htmlSanityCheck/blob/main/src/main/groovy/org/aim42/htmlsanitycheck/report/JUnitXmlReporter.groovy#L90
+  }
+
+  reportSingleCheckDetails (/* SingleCheckResult */ resultForOneCheck) {
+    // empty in original impl: https://github.com/aim42/htmlSanityCheck/blob/main/src/main/groovy/org/aim42/htmlsanitycheck/report/JUnitXmlReporter.groovy#L94
+  }
+
+  closeReport () {
+    // empty in original impl: https://github.com/aim42/htmlSanityCheck/blob/main/src/main/groovy/org/aim42/htmlsanitycheck/report/JUnitXmlReporter.groovy#L98
+  }
+}
+
+module.exports = {
+  createReporter: (/* SinglePageResult[] */ resultsForAllPages, /* ReporterConfig */ config, /* LoggingFacade */ logger) => {
```

Comment:
I think it is better pass resultsForAllPages to the method reportFindings and we have to create a super class for all reporter